parameter/CompoundRule.cpp

```diff
@@ -140,7 +140,9 @@ bool CCompoundRule::matches() const
 bool CCompoundRule::fromXml(const CXmlElement& xmlElement, CXmlSerializingContext& serializingContext)
 {
     // Get type
-    _bTypeAll = xmlElement.getAttributeBoolean("Type", _apcTypes[true]);
+    string strType;
+    xmlElement.getAttribute("Type", strType);
+    _bTypeAll = strType == _apcTypes[true];
```

Comment:
It would be so much nicer to returning the value. Maybe you could take a reference to a bool to warn on error.